library/spdm_common_lib/libspdm_com_opaque_data.c

```diff
@@ -231,13 +225,31 @@ libspdm_process_opaque_data_supported_version_data(libspdm_context_t *spdm_conte
         (opaque_element_support_version->version_count == 0)) {
         return LIBSPDM_STATUS_INVALID_MSG_FIELD;
     }
+
+    /* update the version_count from the received buffer */
+    version_count = opaque_element_support_version->version_count;
+
+    if (data_in_size !=
+        libspdm_get_opaque_data_supported_version_data_size(spdm_context, version_count)) {
+        return LIBSPDM_STATUS_INVALID_MSG_SIZE;
+    }
```

Comment:
@jyao1 
Because, I think this function will also be called to calculate the supported version data size from the received buffer(The requester). And spdm_context->local_context.secured_message_version.spdm_version_count is the version_count from the responder side, It can't cover both cases, Am I right?

Reply:
It does cover both cases.